Formula/overcommit.rb

```diff
@@ -0,0 +1,31 @@
+class Overcommit < Formula
+  desc "Utility to install, configure, and extend Git hooks"
+  homepage "https://github.com/sds/overcommit"
+  url "https://github.com/sds/overcommit/archive/v0.49.1.tar.gz"
+  sha256 "6bf60311211efdd036abe14d62cd314226f99d6c0676810e79df555a72bc9ac3"
+
+  depends_on "ruby"
+
+  resource "childprocess" do
+    url "https://rubygems.org/downloads/childprocess-1.0.1.gem"
+    sha256 "7fe65b8d220967e8b47d8027274db40667641829f12ce9c6145022949c564fb4"
+  end
+
+  resource "iniparse" do
+    url "https://rubygems.org/downloads/iniparse-1.4.4.gem"
+    sha256 "9abf3c9daac0ccaad0829b967e1e98b47ffddba9014295b02fd03d34909971f5"
+  end
+
+  def install
+    ENV["GEM_HOME"] = libexec
+    resources.each do |r|
+      r.fetch
+      system "gem", "install", r.cached_download, "--ignore-dependencies",
+             "--no-document", "--install-dir", libexec
+    end
+    system "gem", "build", "overcommit.gemspec"
+    system "gem", "install", "--ignore-dependencies", "overcommit-#{version}.gem"
+    bin.install libexec/"bin/overcommit"
+    bin.env_script_all_files(libexec/"bin", :GEM_HOME => ENV["GEM_HOME"])
+  end
+end
```

Comment:
Needs a `test do` block.

Reply:
Yeh but what should I be testing?

Reply:
Something that the users would be using too.

Reply:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Overcommit manages git hooks.
I guess I need to create a git repository and commit something?

Reply:
yes, that'd be a good test

Reply:
Creating something that uses a hook is a rather complex test.
I invoked `overcommit --install` and checked that the configuration file exists.
I guess that's good enough but I can do a more extensive test if needed.

Reply:
Ok, I tried to commit the overcommit config file but got:
```
This repository contains hooks installed by Overcommit, but the `overcommit` gem is not installed.
Install it with `gem install overcommit`
```

Which is strange because it is installed but for some reason, cannot be required.

Reply:
The gem's code is installed in: `/home/linuxbrew/.linuxbrew/Cellar/overcommit/0.49.1/libexec/gems/overcommit-0.49.1/lib/`.
Should I install it somewhere else or instruct ruby to be aware of it somehow?